src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1757,31 +1757,42 @@ public static int Min(this IEnumerable<int> source)
         {
             if (source == null) throw Error.ArgumentNull("source");
             int value = 0;
-            bool hasValue = false;
-            foreach (int x in source)
+            
+            using (IEnumerator<int> e = source.GetEnumerator())
```

Comment:
PR https://github.com/dotnet/corefx/pull/2237 has already proposed a bunch of changes to Min/Max, and there's a discussion happening there about the transformations.